Pod/Classes/CollectionViewHelpers.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  CollectionViewHelpers.swift
+//  Pods
+//
+//  Created by Jason Grandelli on 12/6/16.
+//
+//
+
+import Foundation
+
+extension UICollectionView {
+    public final func registerReusableCell<T>(_ cell: ReusableCell<T>) where T: UICollectionViewCell {
+        switch cell {
+        case .class(let identifier):
+            self.register(T.self, forCellWithReuseIdentifier: identifier)
+        case .nib(let identifier, let nibName, let bundle):
+            let nib = UINib(nibName: nibName, bundle: bundle)
+            register(nib, forCellWithReuseIdentifier: identifier)
+        }
+    }
+    
+    public final func dequeueReusableCell<T>(_ cell: ReusableCell<T>, indexPath: IndexPath) -> T where T: UICollectionViewCell {
+        guard let cell = dequeueReusableCell(withReuseIdentifier: cell.identifier, for: indexPath) as? T else {
+            assertionFailure("type error how is this possible?")
+            return T()
```

Comment:
I believe either this should just be `assert()` not `assertionFailure()`.

> `assertionFailure()` have similar name to `assert()`, but it can not be confused. Main difference is that `assertionFailure()` gives a hint to the optimizer that at this point, given context ("if" branch, or function context) ends, making it the `@noreturn` effectively. In debug mode application will terminate, but in release behaviour is undefined.

[Swift Assertions, the Missing Manual](http://blog.krzyzanowskim.com/2015/03/09/swift-asserts-the-missing-manual/#assertionfailure)